### PR TITLE
Fixes gulp watch

### LINKED ISSUE #331: gulp watch command failing when trying to run gulp dev again and again
See the error below. Let me know if anyone else is facing this issue. 
```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: watch /Users/deshraj/Documents/EvalAI/frontend/dist/css ENOENT
    at exports._errnoException (util.js:1022:11)
    at FSWatcher.start (fs.js:1305:19)
    at Object.fs.watch (fs.js:1330:11)
    at Gaze._watchDir (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/gaze.js:289:30)
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/gaze.js:358:10
    at iterate (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:52:5)
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:61:11
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/gaze.js:420:5
    at iterate (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:52:5)
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:61:11
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/gaze.js:420:5
    at iterate (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:52:5)
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:61:11
    at /Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/gaze.js:420:5
    at iterate (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:52:5)
    at Object.forEachSeries (/Users/deshraj/Documents/EvalAI/node_modules/gaze/lib/helper.js:66:3)
[1]-  Killed: 9               gulp watch
```

### PR DESCRIPTION
**Pull request by anweshknayak**
fixed #331 
@deshraj @aka-jain please review

### CODE REVIEW

**Comment by aka-jain** [comment]
@anweshknayak : I think, delete handler should also come with vendor task. Isn't it?

File: gulpfile.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ var gulp = require('gulp'),
     connectModRewrite = require('connect-modrewrite'),
     ngConfig = require('gulp-ng-config'),
     prettyError = require('gulp-prettyerror'),
+    del = require('del'),
```
Comment (aka-jain) [inline gulpfile.js]: @anweshknayak : we are already using del in gulp at line 17. So remove this line please.

**Comment by anweshknayak** [comment]
@aka-jain do I need to create gulp watch for each one of them ( rest remaining) ?

**Comment by aka-jain** [comment]
Yes @anweshknayak .

**Comment by anweshknayak** [comment]
@aka-jain please check

File: gulpfile.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ var gulp = require('gulp'),
     connectModRewrite = require('connect-modrewrite'),
     ngConfig = require('gulp-ng-config'),
     prettyError = require('gulp-prettyerror'),
+    path = require('path'),
```
Comment (aka-jain) [inline gulpfile.js]: Is this dependent on npm? If yes, can you also add the package.json file, I think you forgot to add it.

File: gulpfile.js [hunk 3]
```diff
@@ -48,6 +49,16 @@ gulp.task('vendorjs', function() {
             .pipe(concat(chunkName + '.js'))
             //.on('error', swallowError)
             .pipe(gulp.dest("frontend/dist/vendors"))
+            .pipe(gulp.watch(paths).on('change', function (event){
+                if(event.type == 'deleted'){
+                    var filePathFromSrc = path.relative(path.resolve(paths), event.path);
+                    var destFilePath = path.resolve('frontend/dist/vendors', filePathFromSrc);
+                    del.sync(destFilePath);
+                    }
+                  }))
```
Comment (aka-jain) [inline gulpfile.js]: @anweshknayak : remove these extra lines please.

File: gulpfile.js (hunk 2 above)
Comment (anweshknayak) [inline gulpfile.js]: @aka-jain I don't think it is dependent on npm.

Comment (aka-jain) [inline gulpfile.js]: But then how it gets install, actually every gulp module is installed using npm, and I can see that path is not included in package.json. Is it still working fine?

Comment (anweshknayak) [inline gulpfile.js]: Yes I tried runnning "gulp dev". It is running fine.

Comment (aka-jain) [inline gulpfile.js]: Have you tried to run dev again and again while running watch task as mentioned in issue?

Comment (anweshknayak) [inline gulpfile.js]: Yes it is running fine. Made some small changes. Please check

**Comment by aka-jain** [comment]
@anweshknayak : Ok, I'll check it. Thanks.
@deshraj : Merge it please.

**Pull request merged**